adapters/repos/db/index_queue.go

```diff
@@ -89,14 +92,19 @@ type IndexQueueOptions struct {
 }
 
 type batchIndexer interface {
```

Comment:
it was created because it only needed a subset of the methods of the `VectorIndex` interface, and made it really easy to test the queue logic.
As it's getting bigger and bigger we should eventually get rid of it and use the original one, but the refactoring is not trivial. I'll keep an eye on it and refactor it next time we update the tests.